packages/core/src/amazonq/session/sessionState.ts

```diff
@@ -103,6 +105,20 @@ export abstract class CodeGenBase {
                 case CodeGenerationStatus.COMPLETE: {
                     const { newFileContents, deletedFiles, references } =
                         await this.config.proxyClient.exportResultArchive(this.conversationId)
+
+                    const logFileInfo = newFileContents.find(
+                        (file: { zipFilePath: string; fileContent: string }) =>
+                            file.zipFilePath === RunCommandLogFileName
+                    )
+                    if (logFileInfo) {
+                        logFileInfo.fileContent = truncate(logFileInfo.fileContent, 10000000) // Limit to max 20MB
+                        if (logFileInfo.fileContent.length === 10000000) {
+                            logFileInfo.fileContent += '\n... [truncated]'
```

Comment:
truncate() already does this.
```suggestion
                        logFileInfo.fileContent = truncate(logFileInfo.fileContent, 10000000, '\n... [truncated]') // Limit to max 20MB
```